### DISCUSSION TITLE
Restrict pasting

### DISCUSSION THREAD

**Discussion by Aaditya4i** [mdx-editor/editor#434] [Q&A]
I am trying to restrict pasting in this editor. 

In the case of other textArea or Input, we can either use the inbuilt onPaste property or add an eventListener to prevent the default behaviour of pasting into the input. 

How do I achieve this for MDXEditor ? 

Here is an event listener that I tried to implement, 

```
useEffect(() => {
    const editor = mdxEditorRef.current;
    const handlePaste = (e) => {
      e.preventDefault();
    };
    if (editor) {
      editor.addEventListener("paste", handlePaste);
    }
    return () => {
      if (editor) {
        editor.removeEventListener("paste", handlePaste);
      }
    };
  }, [])
```
  
  I am using the useRef hook to pin point the editor.

**Comment by ta-sakin** [comment]
Facing the same issue. I'm trying to block copy/paste. I tried this it triggers when copy/paste but doesn't block.
``` 
useEffect(() => {
    const handleEvent = (e: ClipboardEvent) => {
      e.preventDefault()
    }

    document.addEventListener('copy', handleEvent)
    document.addEventListener('paste', handleEvent)

    return () => {
      document.removeEventListener('copy', handleEvent)
      document.removeEventListener('paste', handleEvent)
    }
  }, [])
```
